runtime/jcl/common/sun_misc_Unsafe.cpp

```diff
@@ -847,6 +847,85 @@ Java_jdk_internal_misc_Unsafe_objectFieldOffset1(JNIEnv *env, jobject receiver,
 	return offset;
 }
 
+/**
+ * Writes modified memory in an address range from cache to main memory.
+ * Uses memory barriers before and after writeback to ensure ordering.
+ * 
+ * On x86, the writeback is done using the CLWB instruction if available
+ * for performance, falling back to CLFLUSHOPT then CLFLUSH otherwise.
+ * 
+ * @param addr address of block to write back to memory
+ * @param len length of block being written
+ */
+void JNICALL
+Java_jdk_internal_misc_Unsafe_writebackMemory(JNIEnv *env, jobject receiver, jlong addr, jlong len)
+{
+#if defined(J9X86) || defined(J9HAMMER)
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+
+	if (vm->dCacheLineSize > 0) {
+		uintptr_t cache_line_size = vm->dCacheLineSize;
+		U_8 *end_addr = (*(U_8 **) &addr) + (*(uintptr_t *) &len) - 1;
+		U_8 *ptr = (U_8 *)((*(uintptr_t *) &addr) & ~(cache_line_size - 1));
+		if (vm->cpuCacheWritebackCapabilities == J9PORT_X86_FEATURE_CLWB) {
```

Comment:
This should be a switch rather than elseif (the failure can be the default case). The barrier calls could also be moved to above and below the switch.

Reply:
Changed to use a switch.